awscli/autocomplete/serverside/servercomp.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import sys
+import logging
+import jmespath
+
+from awscli.autocomplete.completer import BaseCompleter, CompletionResult
+
+
+LOG = logging.getLogger(__name__)
+
+
+def lazy_call(import_name, **kwargs):
+    """Import a callable and invoke it with the provided kwargs.
+
+    :param import_name: A dotted string of the form ``package.module.Callable``.
+    :type import_name: string
+    :param kwargs: kwargs to pass to the callable once it's imported.
+    :type kwargs:  dict
+
+    """
+    module_name, callable_name = import_name.rsplit('.', 1)
+    __import__(module_name)
+    module = sys.modules[module_name]
+    return getattr(module, callable_name)(**kwargs)
+
+
+class LazyClientCreator(object):
+    """Lazy create a botocore client.
+
+    This class will defer creating a client until the create_client method
+    is invoked.  It will also avoid importing botocore until we need to
+    create a client.
+
+    Importing botocore and creating a botocore session/client is an expensive
+    process, and we only want to do this when we know for sure that we need
+    a client.  This class manages this process.
+
+    """
+    def __init__(self, session_import_name='botocore.session.Session'):
+        self._session = None
+        self._session_import_name = session_import_name
+
+    def create_client(self, service_name, **kwargs):
+        if self._session is None:
+            self._session = self._create_session()
+        return self._session.create_client(service_name, **kwargs)
+
+    def _create_session(self):
+        return lazy_call(self._session_import_name)
+
+
+class ServerSideCompleter(BaseCompleter):
+    def __init__(self, completion_lookup, client_creator):
+        self._completion_lookup = completion_lookup
+        self._client_creator = client_creator
+
+    def complete(self, parsed):
+        """
+
+        :param parsed: The parsed CLI coommand the user has typed.
```

Comment:
I'm tempted to say leave this typo because it sounds like cool and command smashed together.